src/cpu/o3/tdm.cc

```diff
@@ -0,0 +1,257 @@
+      ADD_STAT(fetchLatency,
+               statistics::units::Rate<statistics::units::Count,
+                                       statistics::units::Count>::get(),
+               "Fetch Latency Bound, frontend stalls due to instruction cache "
+               "inefficiency"),
+      ADD_STAT(fetchBandwidth,
+               statistics::units::Rate<statistics::units::Count,
+                                       statistics::units::Count>::get(),
+               "Fetch Bandwidth Bound, frontend stalls due to decoder "
+               "inefficiency")
+{
+    // Frontend L2
+    fetchLatency =
+        fetch->getStats().fetchBubblesMax / (cpu->baseStats.numCycles);
+    fetchBandwidth =
+        cpu->cpuStats.topDownStats.topDownL1.frontendBound - fetchLatency;
+}
+
+TopDownStats::TopDownBadSpeculationL2 ::TopDownBadSpeculationL2(CPU* cpu,
+                                                                Decode* decode,
+                                                                IEW* iew)
+    : statistics::Group(cpu, "TopDownL2_BadSpeculation"),
+      ADD_STAT(branchMissPredicts,
+               statistics::units::Rate<statistics::units::Count,
+                                       statistics::units::Count>::get(),
+               "Branch Miss Predicts"),
+      ADD_STAT(machineClears,
+               statistics::units::Rate<statistics::units::Count,
+                                       statistics::units::Count>::get(),
+               "Memory Order Violations")
+{
+    auto& iewMissPred = iew->getStats().branchMispredicts;
+    auto& decodeMissPred = decode->getStats().branchMispred;
+    auto& memOrderViolations = iew->getStats().memOrderViolationEvents;
+
+    auto brMispredictFraction =
+        (iewMissPred + decodeMissPred) /
+        (iewMissPred + decodeMissPred + memOrderViolations);
+
+    branchMissPredicts = brMispredictFraction *
+                         cpu->cpuStats.topDownStats.topDownL1.badSpeculation;
+
+    machineClears = cpu->cpuStats.topDownStats.topDownL1.badSpeculation -
+                    branchMissPredicts;
+}
+
+TopDownStats::TopDownBackendBoundL2::TopDownBackendBoundL2(CPU* cpu,
+                                                           Rename* rename,
+                                                           IEW* iew)
+    : statistics::Group(cpu, "TopDownL2_BackendBound"),
+      ADD_STAT(memoryBound,
+               statistics::units::Rate<statistics::units::Count,
+                                       statistics::units::Count>::get(),
+               "Memory Bound, backend stalls due to memory subsystem"),
+      ADD_STAT(coreBound,
+               statistics::units::Rate<statistics::units::Count,
+                                       statistics::units::Count>::get(),
+               "Core Bound, backend stalls due to functional unit constraints")
+{
+    // Backend L2
+    executionStalls = (iew->instQueue.getStats().numInstsExec0 -
+                       rename->getStats().idleCycles +
```

Comment:
```suggestion
                       rename->getStats().status[Rename::ThreadStatus::Idle]
```

Due to the new changes in the rename stats this will not work anymore.
https://github.com/gem5/gem5/commit/0525337c970f2c0cc2e0b1d6a2e4d12a475da868
@giactra maybe you can confirm this but I solved it in my code with this.

You also need to remove the two `const` from the getStats() function in rename
```

```

Reply:
Apologies I forgot my change would affect this PR which has higher priority. Yes that is the right solution.